wandb/filesync/upload_job.py

```diff
@@ -63,29 +48,16 @@ def __init__(
         super().__init__()
 
     def run(self) -> None:
-        exception = None
         try:
             self.push()
-        except Exception as e:
```

Comment:
This exception-handling logic now lives in `StepUpload._handle_event`.

A rule of thumb that's served me well is: any `except` block should [log XOR re-raise] the exception.
If an `except` block does neither, then it swallows exceptions without a trace, which is scary; if it does both, then higher-up code will log annoying redundant error messages.

`UploadJob.run` doesn't literally re-`raise` the exception, but it does report the exception back to the parent `StepUpload`, which fits the same mold.